Incorrect type annotation for connect_robust return value

# Incorrect type annotation for connect_robust return value

## Bug Description

The return type annotation for `aio_pika.connect_robust` is currently `AbstractRobustConnection`, but in practice, when passing `connection_class=aio_pika.RobustConnection`, the returned object is an instance of `RobustConnection`. This causes a type mismatch error in static type checkers like mypy when annotating the result as `aio_pika.RobustConnection`, even though the runtime type is correct. The type signature of `connect_robust` does not reflect the actual return type when a specific `connection_class` is provided, leading to false-positive type errors.

## How to Reproduce

I prepared a minimal reproducible example to demonstrate the issue.

```python
# /// script
# requires-python = ">=3.10"
# dependencies = ["aio-pika"]
# ///
import asyncio

import aio_pika


async def loop() -> None:
    connection: aio_pika.RobustConnection = await aio_pika.connect_robust(
        "amqp://guest:guest@127.0.0.1/",
        connection_class=aio_pika.RobustConnection,
    )
    print(type(connection))  # <class 'aio_pika.robust_connection.RobustConnection'>


def main() -> None:
    asyncio.run(loop())


if __name__ == "__main__":
    main()

```

```sh
> uvx mypy main.py
main.py:11: error: Incompatible types in assignment (expression has type "AbstractRobustConnection", variable has type "RobustConnection")  [assignment]
        connection: aio_pika.RobustConnection = await aio_pika.connect_robust(
                                                ^
Found 1 error in 1 file (checked 1 source file)
```

## Expected Behavior

I expect the `connection` variable to have `aio_pika.RobustConnection` type.

--- a/aio_pika/connection.py
+++ b/aio_pika/connection.py
@@ -12,6 +12,7 @@ from typing import (
     Type,
     TypeVar,
     Union,
+    overload,
 )
 
 
@@ -36,6 +37,7 @@ from .tools import CallbackCollection
 
 log = get_logger(__name__)
 T = TypeVar("T")
+ConnectionType = TypeVar("ConnectionType", bound=AbstractConnection)
 
 
 class Connection(AbstractConnection):
@@ -300,6 +302,43 @@ def make_url(
         path="/" + virtualhost,
         query=kw,
     )
+
+
+@overload
+async def connect(
+    url: Union[str, URL, None] = None,
+    *,
+    host: str = "localhost",
+    port: int = 5672,
+    login: str = "guest",
+    password: str = "guest",
+    virtualhost: str = "/",
+    ssl: bool = False,
+    loop: Optional[asyncio.AbstractEventLoop] = None,
+    ssl_options: Optional[SSLOptions] = None,
+    ssl_context: Optional[SSLContext] = None,
+    timeout: TimeoutType = None,
+    client_properties: Optional[FieldTable] = None,
+) -> Connection: ...
+
+
+@overload
+async def connect(
+    url: Union[str, URL, None] = None,
+    *,
+    host: str = "localhost",
+    port: int = 5672,
+    login: str = "guest",
+    password: str = "guest",
+    virtualhost: str = "/",
+    ssl: bool = False,
+    loop: Optional[asyncio.AbstractEventLoop] = None,
+    ssl_options: Optional[SSLOptions] = None,
+    ssl_context: Optional[SSLContext] = None,
+    timeout: TimeoutType = None,
+    client_properties: Optional[FieldTable] = None,
+    connection_class: Type[ConnectionType] = ...,
+) -> ConnectionType: ...
 
 
 async def connect(

--- a/aio_pika/robust_connection.py
+++ b/aio_pika/robust_connection.py
@@ -1,6 +1,6 @@
 import asyncio
 from ssl import SSLContext
-from typing import Any, Optional, Tuple, Type, Union
+from typing import Any, Optional, Tuple, Type, TypeVar, Union, overload
 from weakref import WeakSet
 
 import aiormq.abc
@@ -21,6 +21,11 @@ from .log import get_logger
 from .robust_channel import RobustChannel
 from .tools import CallbackCollection
 
+
+RobustConnectionType = TypeVar(
+    "RobustConnectionType",
+    bound=AbstractRobustConnection,
+)
 
 log = get_logger(__name__)
 
@@ -256,6 +261,43 @@ class RobustConnection(Connection, AbstractRobustConnection):
             self.__reconnection_task = None
 
         return await super().close(exc)
+
+
+@overload
+async def connect_robust(
+    url: Union[str, URL, None] = None,
+    *,
+    host: str = "localhost",
+    port: int = 5672,
+    login: str = "guest",
+    password: str = "guest",
+    virtualhost: str = "/",
+    ssl: bool = False,
+    loop: Optional[asyncio.AbstractEventLoop] = None,
+    ssl_options: Optional[SSLOptions] = None,
+    ssl_context: Optional[SSLContext] = None,
+    timeout: TimeoutType = None,
+    client_properties: Optional[FieldTable] = None,
+) -> RobustConnection: ...
+
+
+@overload
+async def connect_robust(
+    url: Union[str, URL, None] = None,
+    *,
+    host: str = "localhost",
+    port: int = 5672,
+    login: str = "guest",
+    password: str = "guest",
+    virtualhost: str = "/",
+    ssl: bool = False,
+    loop: Optional[asyncio.AbstractEventLoop] = None,
+    ssl_options: Optional[SSLOptions] = None,
+    ssl_context: Optional[SSLContext] = None,
+    timeout: TimeoutType = None,
+    client_properties: Optional[FieldTable] = None,
+    connection_class: Type[RobustConnectionType] = ...,
+) -> RobustConnectionType: ...
 
 
 async def connect_robust(

